test/jaxp/javax/xml/jaxp/unittest/datatype/Bug6937964Test.java

```diff
@@ -158,44 +170,40 @@ public final void testNewDurationYearMonthLexicalRepresentation1() {
         if (DEBUG) {
             System.err.println("DatatypeFactory created: " + datatypeFactory.toString());
         }
+        if (DEBUG) {
+            System.err.println("testing value: \"" + actualLex + "\", expecting: \"" + expectedLex + "\"");
+        }
 
-        // test each value
-        for (int onTestValue = 0; onTestValue < TEST_VALUES_LEXICAL.length; onTestValue = onTestValue + 2) {
+        try {
+            Duration duration = datatypeFactory.newDurationYearMonth(actualLex);
 
             if (DEBUG) {
-                System.err.println("testing value: \"" + TEST_VALUES_LEXICAL[onTestValue] + "\", expecting: \"" + TEST_VALUES_LEXICAL[onTestValue + 1] + "\"");
+                System.err.println("Duration created: \"" + duration.toString() + "\"");
             }
 
-            try {
-                Duration duration = datatypeFactory.newDurationYearMonth(TEST_VALUES_LEXICAL[onTestValue]);
-
-                if (DEBUG) {
-                    System.err.println("Duration created: \"" + duration.toString() + "\"");
-                }
-
-                // was this expected to fail?
-                if (TEST_VALUES_LEXICAL[onTestValue + 1].equals(TEST_VALUE_FAIL)) {
-                    Assert.fail("the value \"" + TEST_VALUES_LEXICAL[onTestValue] + "\" is invalid yet it created the Duration \"" + duration.toString() + "\"");
-                }
+            // was this expected to fail?
+            if (expectedLex.equals(TEST_VALUE_FAIL)) {
+                Assert.fail("the value \"" + actualLex + "\" is invalid yet it created the Duration \"" + duration.toString() + "\"");
+            }
 
-                // right XMLSchemaType?
-                // TODO: enable test, it should pass, it fails with Exception(s)
-                // for now due to a bug
-                try {
-                    QName xmlSchemaType = duration.getXMLSchemaType();
-                    if (!xmlSchemaType.equals(DatatypeConstants.DURATION_YEARMONTH)) {
-                        Assert.fail("Duration created with XMLSchemaType of\"" + xmlSchemaType + "\" was expected to be \""
+            // right XMLSchemaType?
+            // TODO: enable test, it should pass, it fails with Exception(s)
+            // for now due to a bug
+            try {
+                QName xmlSchemaType = duration.getXMLSchemaType();
+                if (!xmlSchemaType.equals(DatatypeConstants.DURATION_YEARMONTH)) {
+                    Assert.fail("Duration created with XMLSchemaType of\"" + xmlSchemaType + "\" was expected to be \""
                                 + DatatypeConstants.DURATION_YEARMONTH + "\" and has the value \"" + duration.toString() + "\"");
-                    }
+                }
                 } catch (IllegalStateException illegalStateException) {
                     // TODO; this test really should pass
                     System.err.println("Please fix this bug that is being ignored, for now: " + illegalStateException.getMessage());
```

Comment:
Do we still have such a bug?

Reply:
Yes, test is still trowing IllegalStateException.